### PR TITLE
JAMES-2586 Methods of inserting mailbox and email change to postgres should allow duplicate pk

### PR DESCRIPTION
**Pull request by hungphan227**

### CODE REVIEW

**Comment by Arsnael** [comment]
Hmmm are you sure about that? Should save really be idempotent for email and mailbox change? Is it the case for other implementations?

**Comment by Arsnael** [comment]
https://ci-builds.apache.org/job/james/job/ApacheJames/job/PR-2267/1/

related?

**Comment by hungphan227** [comment]
> Hmmm are you sure about that? Should save really be idempotent for email and mailbox change? Is it the case for other implementations?

It is idempotent in cassandra implementation

**Comment by hungphan227** [comment]
> https://ci-builds.apache.org/job/james/job/ApacheJames/job/PR-2267/1/
> 
> related?

I don't think so. PostgresBlobStoreDAOTest pass locally

**Comment by Arsnael** [comment]
> I don't think so. PostgresBlobStoreDAOTest pass locally

Rerunning the build but let's see...

**Review by Arsnael — APPROVED** [review]

File: server/data/data-jmap-postgres/src/main/java/org/apache/james/jmap/postgres/change/PostgresMailboxChangeDAO.java [hunk 1]
```diff
@@ -62,6 +63,14 @@ public Mono<Void> insert(MailboxChange change) {
             .set(CREATED, toUUIDArray(change.getCreated()))
             .set(UPDATED, toUUIDArray(change.getUpdated()))
             .set(DESTROYED, toUUIDArray(change.getDestroyed()))
+            .set(DATE, change.getDate().toOffsetDateTime())
+            .onConflictOnConstraint(PRIMARY_KEY_CONSTRAINT_NAME)
```
Comment (vttranlina) [inline server/data/data-jmap-postgres/src/main/java/org/apache/james/jmap/postgres/change/PostgresMailboxChangeDAO.java:67]: The state value is 
`UuidCreator.getTimeOrderedEpoch()`
I don't think it needs to "do Update on Conflict...", It looks like pr tried to fix it for stable testing, not for real-world

Comment (hungphan227) [reply server/data/data-jmap-postgres/src/main/java/org/apache/james/jmap/postgres/change/PostgresMailboxChangeDAO.java:67]: changed to do nothing

**Comment by Arsnael** [comment]
Honestly... not convinced it should. That would omit some notifications.Are we sure it's the behavior we want here? I kinda agree with @vttranlina here, it's just trying to fix IT tests somewhere else, without much thinking of the logic of the code

**Comment by vttranlina** [comment]
maybe pr https://github.com/apache/james-project/pull/2279 related to what we want catch in this pr

**Comment by Arsnael** [comment]
I think we can close this, correct? :)

**Comment by hungphan227** [comment]
> I think we can close this, correct? :)

ok

**Pull request closed without merging**